### ISSUE TITLE
Error during computing consensus step

### ISSUE THREAD

**Issue by rybioinf** [fenderglass/ABruijn#457]
Dear Flye Developers,

I am running into an issue during the (second?) computing concensus step. I am working with a prokaryote genome that is about 5m bp. I have data from a couple different runs, one that is about 12x coverage and the other about 420x coverage. Both jobs fail at the same spot:

[2021-12-09 17:20:01] INFO: >>>STAGE: consensus
[2021-12-09 17:20:01] INFO: Running Minimap2
[2021-12-09 17:25:12] INFO: Computing consensus
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/Users/ryan/miniconda3/lib/python3.9/runpy.py", line 268, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/Users/ryan/miniconda3/lib/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/ryan/miniconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/ryan/documents/Flye_Assembly/Flye/bin/flye", line 25, in <module>
    sys.exit(main())
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/main.py", line 830, in main
    _run(args)
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/main.py", line 583, in _run
    jobs[i].run()
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/main.py", line 322, in run
    consensus_fasta = cons.get_consensus(out_alignment, self.in_contigs,
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/polishing/consensus.py", line 71, in get_consensus
    aln_reader = SynchronizedSamReader(alignment_path, contigs_fasta,
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/utils/sam_parser.py", line 199, in __init__
    self.shared_manager = multiprocessing.Manager()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/context.py", line 57, in Manager
    m.start()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/managers.py", line 553, in start
    self._process.start()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
Traceback (most recent call last):
  File "/Users/ryan/documents/Flye_Assembly/Flye/bin/flye", line 25, in <module>
    sys.exit(main())
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/main.py", line 830, in main
    _run(args)
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/main.py", line 583, in _run
    jobs[i].run()
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/main.py", line 322, in run
    consensus_fasta = cons.get_consensus(out_alignment, self.in_contigs,
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/polishing/consensus.py", line 71, in get_consensus
    aln_reader = SynchronizedSamReader(alignment_path, contigs_fasta,
  File "/Users/ryan/documents/Flye_Assembly/Flye/flye/utils/sam_parser.py", line 199, in __init__
    self.shared_manager = multiprocessing.Manager()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/context.py", line 57, in Manager
    m.start()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/managers.py", line 557, in start
    self._address = reader.recv()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/connection.py", line 255, in recv
    buf = self._recv_bytes()
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/connection.py", line 419, in _recv_bytes
    buf = self._recv(4)
  File "/Users/ryan/miniconda3/lib/python3.9/multiprocessing/connection.py", line 388, in _recv
    raise EOFError
EOFError
[flye.log](https://github.com/fenderglass/Flye/files/7694844/flye.log)

Any advice would be much appreciated.

**Comment by rybioinf** [comment]
Quick update. I have tried to reinstall Flye in Conca (I installed locally the last time) and was unsuccessful due to 'incompatible python specs (I have python 3.9)

(base) ryan@MacBook-Pro-de-Ryan documents % conda install flye
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \ 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                                      

UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - flye -> python[version='2.7.*|>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']

Your python: python=3.9

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

Perhaps this helps explain the problem.
